### PR TITLE
add request-timeout port-forward

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Related downstream bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1543797

Leverages the existing `--request-timeout` flag to add timeout support to `port-forward`.
```
$ kubectl port-forward mypod 8080:8080 --request-timeout=1s
Forwarding from 127.0.0.1:8080 -> 8080
info: timed out request after 1s
```

cc @deads2k

### CODE REVIEW

File: pkg/kubectl/cmd/portforward.go [hunk 1]
```diff
@@ -192,6 +206,16 @@ func (o PortForwardOptions) RunPortForward() error {
 		}
 	}()
 
+	if o.Timeout > 0 {
+		go func() {
+			select {
+			case <-time.After(o.Timeout):
+				fmt.Fprintf(o.errOut, "info: timed out request after %v\n", o.Timeout)
+				o.StopChannel <- struct{}{}
```
Comment (deads2k) [inline pkg/kubectl/cmd/portforward.go]: this looks like its making a time bomb for the command.  Why are you doing this again?

**Comment by deads2k** [comment]
This does not look correct.  Is the connection referenced in the pull being closed by the server or the client?

**Comment by juanvallejo** [comment]
> This does not look correct. Is the connection referenced in the pull being closed by the server or the client?

It is closed by the client. Once we receive a value [from the stopChan here](https://github.com/kubernetes/client-go/blob/master/tools/portforward/portforward.go#L176), the connection is [closed by the client here](https://github.com/kubernetes/client-go/blob/master/tools/portforward/portforward.go#L142).

File: pkg/kubectl/cmd/portforward.go (hunk 1 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/portforward.go]: Per [this bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1543797), there is no way to timeout `kubectl port-forward`, as the `--request-timeout` flag was not being honored

**Comment by deads2k** [comment]
> This does not look correct. Is the connection referenced in the pull being closed by the server or the client?
> It is closed by the client. Once we receive a value from the stopChan here, the connection is closed by the client here.

No, I mean in the bugzilla.  I would expect the server close it.  Even if the server doesn't and the client owns it, this looks highly questionable.

**Comment by juanvallejo** [comment]
@deads2k I found [this related bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1275518) which explains that the node will forcibly timeout streaming connections after the time specified in `streaming-connection-idle-timeout` in the node config.

For the referenced bugzilla in this PR, it is most likely the case that they have this value set at around `5m`.
Will add an explanation in the bugzilla, and close this PR.

**Pull request closed without merging**